sftp-client.c

```diff
@@ -1728,12 +1728,8 @@ do_download(struct sftp_conn *conn, const char *remote_path,
 	}
 	close(local_fd);
 #ifdef WINDOWS
-	if (add_mark_of_web(local_path) == -1) {
-		// (resume_flag ? 0 : O_TRUNC) on line 1355 requires us 
-		// to add the mark of the web after transferring the file. 
-		// if MOTW fails, we need to remove the file before exiting
-		remove(local_path);
-		fatal_f("%s: failed to add mark of the web", local_path);
+	if (motw_zone_id == 5 || add_mark_of_web(local_path) == -1) {
```

Comment:
I think it would be cleaner if the `motw_zone_id` validity check is preformed within `add_mark_of_web` function.